From a6a5b3e6c84c7abf0aa3027868ec3f16dfb0fc55 Mon Sep 17 00:00:00 2001
From: Roland Haas <roland.haas@physics.gatech.edu>
Date: Mon, 7 May 2012 14:59:00 -0700
Subject: [PATCH 07/17] GRHydro_InitData: change 2d array access to avoid
 compiler warnings

* don't access 2d array as 1d array through second index
* always assign something to rho_guess in Bondi solution
---
 src/GRHydro_Bondi.c  |   41 +++++++++++++++++++++++++++++------------
 src/GRHydro_BondiM.c |   39 ++++++++++++++++++++++++++++-----------
 2 files changed, 57 insertions(+), 23 deletions(-)

diff --git a/src/GRHydro_Bondi.c b/src/GRHydro_Bondi.c
index eb98ee4..055e1f7 100644
--- a/src/GRHydro_Bondi.c
+++ b/src/GRHydro_Bondi.c
@@ -123,13 +123,17 @@ static void bl_to_ks_con(CCTK_REAL *x, CCTK_REAL blcon[], CCTK_REAL kscon[] )
         for  Kerr-Schild coordinates.
 
  ******************************************************************************/
-static void dxc_dxs_ks_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
+static void dxc_dxs_ks_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[NDIM][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL r, th, ph; 
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[0][i] = 0. ;
+    }
+  }
   
   r  = x_spher[RR]; 
   th = x_spher[TH];
@@ -165,13 +169,17 @@ static void dxc_dxs_ks_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc
         for  Boyer-Lindquist coordinates.
 
  ******************************************************************************/
-static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
+static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[NDIM][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL r, th, ph, rterm, dr; 
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[j][i] = 0. ;
+    } 
+  }
   
   r  = x_spher[RR]; 
   th = x_spher[TH];
@@ -210,13 +218,17 @@ static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc
         for  "Isotropic" coordinates.
 
  ******************************************************************************/
-static void dxc_dxs_iso_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
+static void dxc_dxs_iso_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[NDIM][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL th, ph,r_iso;
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[j][i] = 0. ;
+    } 
+  }
 
   /* BL spherical coordinates : */
   th = x_spher[TH];
@@ -355,13 +367,17 @@ static void setutcon(CCTK_REAL *vcon, CCTK_REAL gcov[][NDIM])
   ---------------
      -- Covariant Kerr metric in Boyer-Lindquist coordinates. 
 ****************************************************************************/
-static void bl_gcov_func( CCTK_REAL *x, CCTK_REAL gcov[][NDIM])
+static void bl_gcov_func( CCTK_REAL *x, CCTK_REAL gcov[NDIM][NDIM])
 {
   int i,j,k ;
   CCTK_REAL sth,cth,s2,r2,DD,mu ;
   CCTK_REAL r,th;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  gcov[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      gcov[i][j] = 0. ;
+    } 
+  }
 
   r  = x[RR];
   th = x[TH];
@@ -648,7 +664,8 @@ static int find_bondi_solution( CCTK_REAL r, CCTK_REAL *rho, CCTK_REAL *u, CCTK_
 
   if( *rho < 0. ) {  
     if( r > 0.9*rs && r < 1.1*rs ) { 
-      *rho = rhos;
+      rho_guess = rhos;
+      *rho = rho_guess;
     }
     else { 
       //  rhotmp = (sqrt(Qdot) - 1.) * (gamma_eos - 1.) / ( gamma_eos * K );
diff --git a/src/GRHydro_BondiM.c b/src/GRHydro_BondiM.c
index e5e491c..5e556b6 100644
--- a/src/GRHydro_BondiM.c
+++ b/src/GRHydro_BondiM.c
@@ -125,11 +125,15 @@ static void bl_to_ks_con(CCTK_REAL *x, CCTK_REAL blcon[], CCTK_REAL kscon[] )
  ******************************************************************************/
 static void dxc_dxs_ks_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL r, th, ph; 
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[j][i] = 0. ;
+    } 
+  }
   
   r  = x_spher[RR]; 
   th = x_spher[TH];
@@ -165,13 +169,17 @@ static void dxc_dxs_ks_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc
         for  Boyer-Lindquist coordinates.
 
  ******************************************************************************/
-static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
+static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[NDIM][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL r, th, ph, rterm, dr; 
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[j][i] = 0. ;
+    } 
+  }
   
   r  = x_spher[RR]; 
   th = x_spher[TH];
@@ -210,13 +218,17 @@ static void dxc_dxs_bl_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc
         for  "Isotropic" coordinates.
 
  ******************************************************************************/
-static void dxc_dxs_iso_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[][NDIM] )
+static void dxc_dxs_iso_calc(CCTK_REAL *x_cart, CCTK_REAL *x_spher, CCTK_REAL dxc_dxs[NDIM][NDIM] )
 {
-  int i; 
+  int i, j; 
   CCTK_REAL th, ph,r_iso;
   CCTK_REAL sth,cth,sph,cph;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  dxc_dxs[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      dxc_dxs[j][i] = 0. ;
+    } 
+  }
 
   /* BL spherical coordinates : */
   th = x_spher[TH];
@@ -355,13 +367,17 @@ static void setutcon(CCTK_REAL *vcon, CCTK_REAL gcov[][NDIM])
   ---------------
      -- Covariant Kerr metric in Boyer-Lindquist coordinates. 
 ****************************************************************************/
-static void bl_gcov_func( CCTK_REAL *x, CCTK_REAL gcov[][NDIM])
+static void bl_gcov_func( CCTK_REAL *x, CCTK_REAL gcov[NDIM][NDIM])
 {
   int i,j,k ;
   CCTK_REAL sth,cth,s2,r2,DD,mu ;
   CCTK_REAL r,th;
 
-  for( i = 0 ; i < NDIM*NDIM ; i++ ) {  gcov[0][i] = 0. ; } 
+  for( i = 0 ; i < NDIM ; i++ ) {
+    for( j = 0 ; j < NDIM ; j++ ) {
+      gcov[i][j] = 0. ;
+    } 
+  }
 
   r  = x[RR];
   th = x[TH];
@@ -648,7 +664,8 @@ static int find_bondi_solution( CCTK_REAL r, CCTK_REAL *rho, CCTK_REAL *u, CCTK_
 
   if( *rho < 0. ) {  
     if( r > 0.9*rs && r < 1.1*rs ) { 
-      *rho = rhos;
+      rho_guess = rhos;
+      *rho = rho_guess;
     }
     else { 
       //  rhotmp = (sqrt(Qdot) - 1.) * (gamma_eos - 1.) / ( gamma_eos * K );
-- 
1.7.10.4

